pyomo/contrib/pyros/util.py

```diff
@@ -0,0 +1,683 @@
+    '''
+    locally_acceptable = [tc.optimal, tc.locallyOptimal, tc.globallyOptimal]
+    globally_acceptable = [tc.optimal, tc.globallyOptimal]
+    robust_infeasible = [tc.infeasible]
+    try_backups = [tc.feasible, tc.maxTimeLimit, tc.maxIterations, tc.maxEvaluations,
+               tc.minStepLength, tc.minFunctionValue, tc.other, tc.solverFailure,
+               tc.internalSolverError, tc.error,
+               tc.unbounded, tc.infeasibleOrUnbounded, tc.invalidProblem, tc.intermediateNonInteger,
+               tc.noSolution, tc.unknown]
+    aborted = [tc.userInterrupt, tc.resourceInterrupt, tc.licensingProblems]
+
+    termination_condition = results.solver.termination_condition
+    if config.solve_master_globally == False:
+        if termination_condition in locally_acceptable:
+            return (False, None)
+        elif termination_condition in robust_infeasible:
+            return (False, grcsTerminationCondition.robust_infeasible)
+        elif termination_condition in try_backups:
+            return (True, None)
+        else:
+            raise NotImplementedError("This solver return termination condition (%s) "
+                                      "is currently not supported by PyROS." % termination_condition)
+    else:
+        if termination_condition in globally_acceptable:
+            return (False, None)
+        elif termination_condition in robust_infeasible:
+            return (False, grcsTerminationCondition.robust_infeasible)
+        elif termination_condition in try_backups:
+            return (True, None)
+        else:
+            raise NotImplementedError("This solver return termination condition (%s) "
+                                      "is currently not supported by PyROS." % termination_condition)
+
+
+def output_logger(config, **kwargs):
+    '''
+    All user returned messages (termination conditions, runtime errors) are here
+    Includes when
+    "sub-solver %s returned status infeasible..."
+    :return:
+    '''
+
+    # === PREAMBLE + LICENSING
+    # Version printing
+    if "preamble" in kwargs:
+        if kwargs["preamble"]:
+            version = str(kwargs["version"])
+            preamble = "\n" \
+                       "===========================================================================================\n" \
+                       "PyROS: Pyomo Robust Optimization Solver v.%s \n" \
+                       "Developed by Natalie M. Isenberg (1), John D. Siirola (2), Chrysanthos E. Gounaris (1) \n" \
+                       "(1) Carnegie Mellon University, Department of Chemical Engineering \n" \
+                       "(2) Sandia National Laboratory, Discrete Math and Optimization Department\n\n" \
+                       "The developers gratefully acknowledge support from the U.S. Department of Energy's \n" \
+                       "Institute for the Design of Advanced Energy Systems (IDAES) \n" \
+                       "Licensing: TODO \n" \
```

Comment:
I would remove the "Licensing: TODO".  All of Pyomo is licensed under the BSD license.